cpp/include/cudf_test/stream_checking_resource_adaptor.hpp

```diff
@@ -57,12 +58,26 @@ class stream_checking_resource_adaptor final : public rmm::mr::device_memory_res
   stream_checking_resource_adaptor& operator=(stream_checking_resource_adaptor&&) noexcept =
     default;
 
+  /**
+   * @brief Return rmm::device_async_resource_ref of the upstream resource.
+   *
+   * @return rmm::device_async_resource_ref of the upstream resource.
```

Comment:
```suggestion
   * @brief Returns the upstream resource for this resource
   *
   * @return Reference to the upstream resource
```
The type should not be included in the tag.
https://github.com/rapidsai/cudf/blob/branch-24.04/cpp/doxygen/developer_guide/DOCUMENTATION.md#return

Reply:
Hm it is not a reference, but a wrapper around the resource

Maybe just go with
```suggestion
   * @brief Returns the wrapped upstream resource
   *
   * @return The wrapped upstream resource
```